Defining the deadline in hypothesis

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32143 Defining the deadline in hypothesis**

Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)

Commit:
Defining the deadline in hypothesis

z-a-f commented on test/hypothesis_utils.py:
```diff
@@ -310,14 +310,18 @@ def tensor_conv(
+settings._settings__definitions_are_locked = True
 current_settings = settings._profiles[settings._current_profile].__dict__
 current_settings['deadline'] = None
-if hypothesis_version >= (3, 16, 0) and hypothesis_version < (5, 0, 0):
```
There is no need to set the timeout to unlimited -- it was unlimited by default in all versions

z-a-f commented:
This change is needed because the `settings()` uses the default profile not the current one. This change registers the `deadline` for all profiles.

Commit:
Update on "Defining the deadline in hypothesis"


Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)

Commit:
Update on "Defining the deadline in hypothesis"


Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 2278de31:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=d97413eb7ae34355022c47da1dd5c8eba2a220f7&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base d97413eb since Jan 13
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  Since your merge base is older than `viable/strict`, run these commands:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=2278de31c4d1e658d1b75b91656f0e1002d53a73).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=84799688) from Jan 13 until Jan 13



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "Defining the deadline in hypothesis"


Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)

Commit:
Update on "Defining the deadline in hypothesis"


Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)

Commit:
Update on "Defining the deadline in hypothesis"


Differential Revision: [D19385221](https://our.internmc.facebook.com/intern/diff/D19385221)
